### ISSUE TITLE
Add test for collation in BigQuery connector

### ISSUE THREAD

**Issue by ebyhr** [trinodb/trino#14901]
https://cloud.google.com/bigquery/docs/reference/standard-sql/collation-concepts

**Comment by tan31989** [comment]
@ebyhr Can I work on this and create a PR?

**Comment by ebyhr** [comment]
@tan31989 Sure, please go for it! You can refer to collate tests in Postgres connector.

**Comment by tan31989** [comment]
Great, thanks @ebyhr for confirming. Will start working on this.

**Comment by ebyhr** [comment]
@tan31989 Are you still working on this?

**Comment by tan31989** [comment]
Yes @ebyhr, will raise a PR for this in some days. Got delayed because of other priorities

**Comment by tan31989** [comment]
@ebyhr I was trying to execute this locally via my personal GCP account. Followed the README mentioned in bigquery module. But I keep seeing that the table load was not successful with initialisation like:

```sh
java.lang.AssertionError: [Table is not loaded properly: tpch.tiny.orders] 
Expecting:
 <15000L>
to be equal to:
 <0L>
but was not.
```

There doesn't seem to be any specific error while I debug at [this](https://github.com/trinodb/trino/blob/master/plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/BigQueryQueryRunner.java#L91) line. Basically, the SQL to load doesn't load the data 🤔 

Am I missing any other setting?

**Comment by hashhar** [comment]
Sorry about that @tan31989. There's a bug where tables written to BigQuery don't immediately return correct results for `count(*)` queries. See https://github.com/trinodb/trino/issues/14981.

I have a fix at https://github.com/trinodb/trino/pull/15542 - you can try to work on top of that branch in meantime or comment out https://github.com/trinodb/trino/blob/fcd603d31ef43f43dab5e1664a43b9b655cc8195/testing/trino-testing/src/main/java/io/trino/testing/QueryAssertions.java#L342 for now.

**Comment by tan31989** [comment]
Thanks @hashhar for the quick response and update.

**Comment by tan31989** [comment]
@ebyhr need some input on this, so while BigQuery supports collate function, I see that a basic test fails in the trino as `TrinoException: line 1:8: Function 'collate' not registered`

Is there a reference on how to register functions?

**Comment by ebyhr** [comment]
You need to create tables on BigQuery instead of Trino.

**Comment by tan31989** [comment]
Maybe I am wrong here, but isn't it the idea to test the execution of SQL via trino? 
Meaning, if I want a simple collate SQL to be executed in BigQuery, I should be able to execute this as:

`assertQuery("SELECT COLLATE('a', 'und:ci') AS character", "VALUES ('a')");` ?

**Comment by ebyhr** [comment]
I mean creating a table having COLLATE definition on BigQuery and then reading the table on Trino.
